ra2ce/analysis/adaptation/adaptation_option.py

```diff
@@ -39,71 +51,115 @@ class AdaptationOption:
-        damages_section: AnalysisSectionDamages,
-        losses_section: AnalysisSectionLosses,
     ) -> AdaptationOption:
-        # Adjust path to the input files
-        def extend_path(analysis: str, input_path: Path) -> Path:
-            if not input_path:
+        """
+        Classmethod to create an adaptation option from an analysis configuration and an adaptation option.
+
+        Args:
+            analysis_config (AnalysisConfigWrapper): Analysis config input
+            adaptation_option (AnalysisSectionAdaptationOption): Adaptation option input
+
+        Raises:
+            ValueError: If damages and losses sections are not present in the analysis config data.
+
+        Returns:
+            AdaptationOption: The created adaptation option.
+        """
+        # Adjust path to the files (should be in Adaptation/input)
+        def construct_path(
+            orig_path: Path,
+            analysis: str,
+            folder: str,
+        ) -> Path:
+            if not orig_path:
                 return None
-            # Input is directory: add stuff at the end
-            if not (input_path.suffix):
-                return input_path.joinpath("input", adaptation_option.id, analysis)
-            return input_path.parent.joinpath(
-                "input", adaptation_option.id, analysis, input_path.name
+            # Orig is directory: add stuff at the end
+            if not (orig_path.suffix):
+                return orig_path.parent.joinpath(
+                    "input", adaptation_option.id, analysis, folder
+                )
+            return orig_path.parent.parent.joinpath(
+                "input", adaptation_option.id, analysis, folder, orig_path.name
+            )
+
+        def replace_paths(
+            config_data: AnalysisConfigData, analysis: str
+        ) -> AnalysisConfigData:
+            config_data.input_path = construct_path(
+                config_data.input_path, analysis, "input"
             )
+            config_data.static_path = construct_path(
+                config_data.static_path, analysis, "static"
+            )
+            config_data.output_path = construct_path(
+                config_data.output_path, analysis, "output"
+            )
+            return config_data
 
-        if not damages_section or not losses_section:
+        if (
+            not analysis_config.config_data.damages_list
+            or not analysis_config.config_data.losses_list
+        ):
             raise ValueError(
                 "Damages and losses sections are required to create an adaptation option."
             )
 
-        _damages_root = extend_path("damages", root_path)
-        _damages_section = deepcopy(damages_section)
+        # Create input for the damages analysis
+        _damages_config = deepcopy(analysis_config)
+        _damages_config.config_data = replace_paths(
+            _damages_config.config_data, "damages"
+        )
```

Comment:
is this replacing of path an assumption?

Reply:
Now use the `config_value` of the enum.